eodatasets3/model.py

```diff
@@ -373,6 +372,81 @@ def producer_abbreviated(self) -> Optional[str]:
             )
 
 
+class ComplicatedNamingConventionsAlchemist(ComplicatedNamingConventions):
+
+    """
+    This class is inherited from ComplicatedNamingConventions
+    and overrides few attributes specific to C3 dataprocessing at the Alchemist end.
+    """
+
+    @classmethod
+    def for_c3_processing(cls, dataset: EoFields, uri=DEA_URI_PREFIX):
+        """
+        The required fields for the c3 data processing are controlled here.
+        """
+        return cls(
+            dataset=dataset,
+            base_product_uri=uri,
+            required_fields=(
+                "eo:platform",
+                "odc:dataset_version",
+                "odc:collection_number",
+                "odc:processing_datetime",
+                "odc:producer",
+                "odc:product_family",
+                "odc:region_code",
+                "dea:dataset_maturity",
+            ),
+        )
+
+    @property
+    def _org_collection_number(self) -> Optional[int]:
+
+        # Deliberately fail if collection_number is not defined in the config yaml
+        return int(self.dataset.collection_number)
+
+    def _product_group(self, subname=None) -> str:
+        # Computues product group, e.g "ga_ls_wo_3"
+        # Deliberately fail if any of these attributes not found.
+        parts = [
+            self.producer_abbreviated,
+            self.platform_abbreviated,
+            self.dataset.product_family,
+        ]
+        return "_".join(parts)
+
+    def destination_folder(self, base: Path):
+        self._check_enough_properties_to_name()
+        parts = [self.product_name, self.dataset.dataset_version.replace(".", "-")]
+        parts.extend(utils.subfolderise(self.dataset.region_code))
+        parts.extend(f"{self.dataset.datetime:%Y/%m/%d}".split("/"))
+        return base.joinpath(*parts)
+
+    def _dataset_label(self, sub_name: str = None):
+        # Copycat of the parent method except version is simply collection number
+        p = self.dataset
+        version = p.collection_number
+        maturity: str = p.properties.get("dea:dataset_maturity")
+        return "_".join(
+            [
+                p
+                for p in (
+                    self._product_group(sub_name),
+                    version,
```

Comment:
I don't like this, it's misleading.

The join should just be the `product_group`, which really is `product_id` and then all the other bits. Including version here is not correct.

Reply:
Spot on 👍 simplified it.